build/templates/session.py.mako

```diff
@@ -297,6 +315,15 @@ class Session(_SessionBase):
 %       endfor
         self._param_list = ', '.join(param_list)
 
+        # self.get_channel_names() and self.channel_count can only be called after the session
+        # handle `self._${config['session_handle_parameter_name']}` is set
+        try:
+            self._all_channels_in_session = self.get_channel_names(range(self.channel_count))
+        except AttributeError:
+            self._all_channels_in_session = None
+
+        # _is_frozen must be set last to prevent __setattr__ from raising an exception while
```

Comment:
suggestion:

Finally, set _is_frozen to True which is used to prevent clients from accidentally adding members when trying to set a property with a typo.

(describe the true reason, not the symptom of not doing it correctly)